File: its-my-training-day/leetcode-top-75/questions/arrays-hashing/heap/kth-largest-in-array/solution.cpp
#include "../../../../shared/cpp/ds.hpp"

#include <algorithm>
#include <cctype>
#include <cstdint>
#include <functional>
#include <string>
#include <unordered_map>
#include <vector>

int kthLargestInArray(const std::vector<int>& nums, int k) {
    MinHeap<int> heap;
    for (int i = 0; i < static_cast<int>(nums.size()); ++i) {
        heap.push(nums[i]);
        if (static_cast<int>(heap.size()) > k) heap.pop();
    }
    return heap.peek();
}

File: its-my-training-day/leetcode-top-75/questions/arrays-hashing/heap/kth-largest-in-array/solution.ts
import {
  ArrayList,
  Stack,
  Queue,
  ListNode,
  TreeNode,
  MinHeap,
  MaxHeap,
} from "../../../../shared/typescript/ds";

export const kthLargestInArray = (nums: number[], k: number): number => {
  const heap = new MinHeap<number>();
  for (let i = 0; i < nums.length; i += 1) {
    heap.push(nums[i]);
    if (heap.size() > k) heap.pop();
  }
  return heap.peek() as number;
};
